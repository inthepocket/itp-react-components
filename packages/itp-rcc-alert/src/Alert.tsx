import * as React from 'react';
import classnames from 'classnames';
import capitalize from 'capitalize';
import { default as Button } from '@inthepocket/itp-rcc-button';

const Alert = ({
  children,
  dismissButtonAriaLabel,
  dismissIcon,
  hasDismissButton,
  onDismiss,
  prefixCss,
  styles,
  type,
}: AlertPropsInterface) => {
  const containerClassName = styles ? styles.alert : prefixCss;
  const dissmissClassName = styles ? styles.isDismissible : `${prefixCss}--dismissible`;
  const typeClassName = styles ? styles[`type${capitalize(type)}`] : `${prefixCss}-type-${type}`;
  const dismissPrefixCss = styles ? 'dismissButton' : `${prefixCss}-dismiss-button`;

  const className = classnames({
    [`${containerClassName}`]: containerClassName,
    [`${typeClassName}`]: typeClassName,
    [`${dissmissClassName}`]: hasDismissButton && dissmissClassName,
  });

  return (
    <div className={className}>
      {children}
      {hasDismissButton && (
        <Button aria-label={dismissButtonAriaLabel} onClick={onDismiss} prefixCls={dismissPrefixCss} styles={styles}>
          {dismissIcon}
        </Button>
      )}
    </div>
  );
};

export interface AlertStylesInterface {
  alert: string;
  [key: string]: string | undefined;
}

export interface AlertPropsInterface {
  children: React.ReactNode | Element;
  dismissIcon?: React.ReactNode | Element;
  dismissButtonAriaLabel?: string;
  hasDismissButton?: Boolean;
  onDismiss?: Function;
  prefixCss: string;
  styles?: AlertStylesInterface;
  type?: 'default' | 'success' | 'danger' | 'warning' | 'info';
};

Alert.defaultProps = {
  dismissButtonAriaLabel: 'Dismiss',
  dismissIcon: <span aria-hidden="true">&times;</span>,
  hasDismissButton: false,
  onDismiss: null,
  prefixCss: 'alert',
  type: 'default',
};

export default Alert;
