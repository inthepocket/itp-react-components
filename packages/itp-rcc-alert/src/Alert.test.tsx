import * as React from 'react';
import { create } from 'react-test-renderer';
import Alert from './Alert';

const cssModuleMock = {
  alert: 'cssModuleAlert',
  dismissButton: 'cssModuleDismissButton',
  typeDefault: 'cssModuleTypeDefault',
  typeSuccess: 'cssModuleTypeSuccess',
  typeDanger: 'cssModuleTypeDanger',
  typeInfo: 'cssModuleTypeInfo',
};

test('Alert: default', () => {
  const tree = create((
    <Alert>
      <p>Default alert</p>
    </Alert>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});
