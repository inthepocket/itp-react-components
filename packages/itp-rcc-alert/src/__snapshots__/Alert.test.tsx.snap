// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert: danger 1`] = `
<div
  className="alert alert-type-danger"
>
  <p>
    Danger alert
  </p>
</div>
`;

exports[`Alert: default 1`] = `
<div
  className="alert alert-type-default"
>
  <p>
    Default alert
  </p>
</div>
`;

exports[`Alert: info 1`] = `
<div
  className="alert alert-type-info"
>
  <p>
    Info alert
  </p>
</div>
`;

exports[`Alert: success 1`] = `
<div
  className="alert alert-type-success"
>
  <p>
    Success alert
  </p>
</div>
`;

exports[`Alert: warning 1`] = `
<div
  className="alert alert-type-warning"
>
  <p>
    Warning alert
  </p>
</div>
`;

exports[`Alert: with css module: danger 1`] = `
<div
  className="cssModuleAlert cssModuleTypeDanger"
>
  <p>
    Danger alert
  </p>
</div>
`;

exports[`Alert: with css module: default 1`] = `
<div
  className="cssModuleAlert cssModuleTypeDefault"
>
  <p>
    Default alert
  </p>
</div>
`;

exports[`Alert: with css module: info 1`] = `
<div
  className="cssModuleAlert cssModuleTypeInfo"
>
  <p>
    Info alert
  </p>
</div>
`;

exports[`Alert: with css module: success 1`] = `
<div
  className="cssModuleAlert cssModuleTypeSuccess"
>
  <p>
    Success alert
  </p>
</div>
`;

exports[`Alert: with css module: warning 1`] = `
<div
  className="cssModuleAlert cssModuleTypeWarning"
>
  <p>
    Warning alert
  </p>
</div>
`;

exports[`Alert: with css module: with dismiss button 1`] = `
<div
  className="cssModuleAlert cssModuleTypeDefault"
>
  <p>
    Dismissible alert
  </p>
  <button
    aria-label="Dismiss"
    className="cssModuleDismissButton"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;

exports[`Alert: with css module: with dismiss button and dismiss icon 1`] = `
<div
  className="cssModuleAlert cssModuleTypeDefault"
>
  <p>
    Dismissible alert
  </p>
  <button
    aria-label="Dismiss"
    className="cssModuleDismissButton"
    onClick={[Function]}
    type="button"
  >
    <span>
      dismiss icon
    </span>
  </button>
</div>
`;

exports[`Alert: with css module: with dismiss button, dismiss icon and aria label 1`] = `
<div
  className="cssModuleAlert cssModuleTypeDefault"
>
  <p>
    Dismissible alert
  </p>
  <button
    aria-label="Dismiss label"
    className="cssModuleDismissButton"
    onClick={[Function]}
    type="button"
  >
    <span>
      dismiss icon
    </span>
  </button>
</div>
`;

exports[`Alert: with dismiss button 1`] = `
<div
  className="alert alert-type-default alert--dismissible"
>
  <p>
    Dismissible alert
  </p>
  <button
    aria-label="Dismiss"
    className="alert-dismiss-button"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;

exports[`Alert: with dismiss button and dismiss icon 1`] = `
<div
  className="alert alert-type-default alert--dismissible"
>
  <p>
    Dismissible alert
  </p>
  <button
    aria-label="Dismiss"
    className="alert-dismiss-button"
    onClick={[Function]}
    type="button"
  >
    <span>
      dismiss icon
    </span>
  </button>
</div>
`;

exports[`Alert: with dismiss button, dismiss icon and aria label 1`] = `
<div
  className="alert alert-type-default alert--dismissible"
>
  <p>
    Dismissible alert
  </p>
  <button
    aria-label="Dismiss label"
    className="alert-dismiss-button"
    onClick={[Function]}
    type="button"
  >
    <span>
      dismiss icon
    </span>
  </button>
</div>
`;
